### PR TITLE
tests: Use mock kn client tests for service list and delete

### LINKED ISSUE #358: Refactor service unit tests to use KnClient mock framework
## Describe the feature:

In order to simplify the maintenance of our tests, existing tests in services should be refactored to use the `MockKnClient` support class. An example of such a test can be found in [`service_create_mock_test.go`](https://github.com/knative/client/blob/master/pkg/kn/commands/service/service_create_mock_test.go)

Ideally all the test classes

* service_create_test.go
* service_list_test.go
* service_describe_test.go
* service_update_test.go

should be rewritten (and the mock sample integrated into service_create_test)

Maybe also add as service_delete_test.go which seems to be missing.

### PR DESCRIPTION
**Pull request by itsmurugappan**
Fixes #358 

## Proposed Changes

* Refactors the delete service and list service unit tests to use knclient mock
* Partially fixes #358

### CODE REVIEW

File: pkg/kn/commands/service/list_mock_test.go [hunk 1]
```diff
@@ -82,6 +82,146 @@ func TestServiceListAllNamespaceMock(t *testing.T) {
...
+	r.ListServices(knclient.Any(), &v1alpha1.ServiceList{}, nil)
+
+	output, err := executeServiceCommand(client, "list")
+	assert.NilError(t, err)
+	assert.Assert(t, util.ContainsAll(output, "No", "services", "found"))
+
+	r.Validate()
+}
+
+func TestGetEmptyMock(t *testing.T) {
```
Comment (navidshaikh) [inline pkg/kn/commands/service/list_mock_test.go]: should we rename this to `TestListEmptyWithArgMock` ?

File: pkg/kn/commands/service/list_mock_test.go [hunk 2]
```diff
@@ -82,6 +82,146 @@ func TestServiceListAllNamespaceMock(t *testing.T) {
...
+	outputLines := strings.Split(output, "\n")
+	assert.Check(t, util.ContainsAll(outputLines[0], "NAME", "URL", "LATEST", "AGE", "CONDITIONS", "READY", "REASON"))
+	assert.Check(t, util.ContainsAll(outputLines[1], "bar", "bar.default.example.com", "bar-xyz"))
+	assert.Check(t, util.ContainsAll(outputLines[2], "foo", "foo.default.example.com", "foo-xyz"))
+	assert.Check(t, util.ContainsAll(outputLines[3], "sss", "sss.default.example.com", "sss-xyz"))
+
+	r.Validate()
+}
+
+func TestServiceListAllNamespacesOutputMock(t *testing.T) {
```
Comment (navidshaikh) [inline pkg/kn/commands/service/list_mock_test.go]: This seems redundant with https://github.com/knative/client/blob/master/pkg/kn/commands/service/list_mock_test.go#L31

File: pkg/kn/commands/service/list_mock_test.go [hunk 3]
```diff
@@ -82,6 +82,146 @@ func TestServiceListAllNamespaceMock(t *testing.T) {
...
+
+	outputLines := strings.Split(output, "\n")
+	assert.Check(t, util.ContainsNone(outputLines[0], "NAME", "URL", "LATEST", "AGE", "CONDITIONS", "READY", "REASON"))
+	assert.Check(t, util.ContainsAll(outputLines[0], "bar", "bar.default.example.com", "bar-xyz"))
+	assert.Check(t, util.ContainsAll(outputLines[1], "foo", "foo.default.example.com", "foo-xyz"))
+
+	r.Validate()
+}
+
+func TestServiceGetOneOutputMock(t *testing.T) {
```
Comment (navidshaikh) [inline pkg/kn/commands/service/list_mock_test.go]: *TestServiceListOneOutputMock

i.e. `s/Get/List`

File: pkg/kn/commands/service/list_mock_test.go (hunk 2 above)
Comment (itsmurugappan) [reply pkg/kn/commands/service/list_mock_test.go]: removed the redundant method

File: pkg/kn/commands/service/list_mock_test.go (hunk 3 above)
Comment (itsmurugappan) [reply pkg/kn/commands/service/list_mock_test.go]: renamed the func to TestServiceListOneOutputMock

File: pkg/kn/commands/service/list_mock_test.go (hunk 1 above)
Comment (itsmurugappan) [reply pkg/kn/commands/service/list_mock_test.go]: yes. makes sense. renamed it

**Pull request merged**